actionview/lib/action_view/helpers/tag_helper.rb

```diff
@@ -70,6 +70,7 @@ def tag_string(name, content = nil, escape: true, **options, &block)
           content = @view_context.capture(self, &block) if block_given?
           self_closing = SVG_SELF_CLOSING_ELEMENTS.include?(name)
           if (HTML_VOID_ELEMENTS.include?(name) || self_closing) && content.nil?
+            TagHelper.ensure_valid_html5_tag_name(name.to_s)
             "<#{name.to_s.dasherize}#{tag_options(options, escape)}#{self_closing ? " />" : ">"}".html_safe
```

Comment:
As part of https://github.com/rails/rails/commit/649516ce0feb699ae06a8c5e81df75d460cc9a85, the keyword argument for `tag_string` was renamed from `escape_attributes:` to `escape:` (since the commit added support for element name escaping).

Now that we're removing element name escaping in favor of failing fast, would it make sense to revert the keyword option back to `escape_attributes:`?

https://github.com/rails/rails/commit/649516ce0feb699ae06a8c5e81df75d460cc9a85 is part of the `7.1.0.alpha`, so the keyword change has not yet been part of a release.

While it might not have an impact in most circumstances, the key change is a breaking change, and reverting could reduce the number of breaking changes in the release by one.

Since `escape:` (and, to be fair, `escape_attributes:`) are special-case keyword arguments, that means that calls to `tag`-prefixed builder methods can never render an `[escape]` HTML attribute. For example, a call to `tag.my_custom_element escape: true` wouldn't render `<my-custom-element escape="true"><my-custom-element>`, it'd render `<my-custom-element></my-custom-element>`. The same would be true for `escape_attributes:`, but that feels less likely to be an HTML attribute name.

Currently, callers need to know about the special-casing `escape:` option, and pass the value as `"escape" => true`.

Reply:
@seanpdoyle I got your point. Is it fine to create an another PR for renaming the attribute. I'm happy to work on that :)

Reply:
@akhilgkrishnan a follow-up PR makes sense to me! I don't have much authority on the outcome 😄. I'm mostly just posing the question for the reviewers who will ultimately merge in your change.

Reply:
Yeah, that totally makes sense 😆

Reply:
@seanpdoyle As we are using `escape` in other methods as well. So is it worth for changing back  to `escape_attributes:` only for `tag_string`

Reply:
I think since the interface change was shipped as part of 7.1, the breaking change is set in stone. Probably not worth reverting now.

Reply:
Yes